core/src/main/scala/akka/persistence/jdbc/util/ConfigOps.scala

```diff
@@ -55,11 +56,15 @@ object ConfigOps {
 
     def ?[A](key: String): Try[A] = as(key)
 
-    def ?:[A](key: String, default: A) = as(key, default)
+    def ?:[A](key: String, default: A) = as(key, noDefault(key))
 
     def withkey[A](key: String)(f: Config => A): A = f(config.getConfig(key))
   }
 
+  private def noDefault(key: String) = {
+    throw new ConfigurationException(s"The [$key] setting relies on a default provided in code.")
+  }
```

Comment:
I think it would be better to use the same approach as we use in other Akka libraries, i.e. no such Try stuff if it's reading a config property that is expected to be defined in reference.conf.
But this might be a step in the right direction.

Reply:
👍 double 

Great to get rid of the double defaults

Reply:
Yes, the possibility to put in defaults should be removed altogether.
So far this PR is mainly to show what was relying on these defaults.